src/vs/workbench/browser/parts/editor/editorPart.ts

```diff
@@ -1143,6 +1178,12 @@ export class EditorPart extends Part implements IEditorPart {
 
 		this._onDidChangeSizeConstraints.input = gridWidget.onDidChange;
 		this._onDidScroll.input = gridWidget.onDidScroll;
+		this.gridWidgetDisposables.clear();
+		this.gridWidgetDisposables.add(gridWidget.onDidChangeViewMaximized(maximized => this._onDidChangeGroupMaximized.fire(maximized)));
+
+		if (this.activeGroup && this.isGroupMaximized(this.activeGroup)) {
+			this._onDidChangeGroupMaximized.fire(true);
+		}
 
```

Comment:
I am still not so happy with the check for `this.activeGroup` here: isnt it that the grid widget can have one of its views maximized or not? If so, can we not simply have a method on the widget itself if any view is maximized and stop the need for an editor group to figure this out?

Besides, I think there is also a bug: if the current grid widget has a group maximized and the new one not, we do not fire an event. To reproduce:
- have a 2x2 layout
- maximize the group bottom left
- apply the 2x2 layout again from command palette or view menu
- the maximize button is not removed

Reply:
I aligned it with the way isGroupExpanded works, which also has the same principal of only one group being expanded at once. But, I'm fine with changing it.